docs/using-third-party-react-components.md

```diff
@@ -0,0 +1,90 @@
+# Using third party React components
+
+Using a third party (Javascript) React component is straightforward for most components. There are two ways of declaring a third party React component in F# - either by declaring a Pojo record for the props or by declaring a Discriminated Union where each case has one field.
+
+Some components have a [Typescript](https://www.typescriptlang.org/) definition available, either because the component was authored in Typescript or someone created a type definition for the [Definitely Typed project](https://definitelytyped.org/). If this is the case then you can try the [ts2fable tool](https://github.com/fable-compiler/ts2fable) to convert this React component type definition from Typescript to a Fable type declaration - it might need some tweaking but for components with a big API surface this can be a real time safer.
+
+The basic steps when working with a Discriminated Union are:
+
+### 1. Install the react component
+
+Using yarn or npm, install the react component you want to use.
+
+For example to use the [rc-progress React components](https://github.com/react-component/progress), run the following in your Fable project root:
+
+```bash
+yarn install rc-progress
+```
+
+### 2. Define the props type
+
+Reference the **documentation of the React component** to find out which props the component supports and declare them as an F# type (see below for the two supported mechanisms). You can define only a subset of supported props in F# if you don't need to cover the full props options that the React component supports.
+
+For example to expose the percent, strokeWidth and strokeColor props of the rc-progress components:
+
+```fsharp
+type ProgressProps =
+  | Percent of int
+  | StrokeWidth of int
+  | StrokeColor of string
+```
+
+If one of the props is treated as a string enum in Javascript (e.g. if there is a size prop with the supported values "small", "normal" and "big"), then the `[<StringEnum>]` attribute can be very useful for defining helper types:
```

Comment:
Maybe adding a link to StringEnum in Fable docs? http://fable.io/docs/interacting.html#stringenum-attribute